src/controllers/confirm.company.controller.ts

```diff
@@ -17,10 +18,36 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
 };
 
 export const post = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const company: CompanyProfile = await getCompanyProfile(req.query.companyNumber as string);
+    if (isStatusEligible(company)) {
+      create(req);
+      const url = CONFIRMATION_STATEMENT_TRADING_STATUS.replace(":companyNumber", company.companyNumber);
+      return res.redirect(url);
+    } else {
+      return res.render(Templates.INVALID_STATUS);
+    }
+  } catch (e) {
+    next(e);
+  }
+};
+
+const isStatusEligible = (company: CompanyProfile): boolean => {
+  const status: string = company.companyStatus;
```

Comment:
the web shouldn't have any knowledge on what companies are eligible or not, that should be entirely contained to the api.
Which unfortunately would mean for this story to be accomplished the web will first need to create a transaction against a company.

So the flow will need to be confirm and continue -> redirect to auth page -> call api to check eligibility (for this story it will post internalapi/transaction/id/confirmation-statement/ -> render approproate stop screen

Reply:
It sounds like this needs more discussion to me, we really should settle on a plan for this.